weave/trace_server/errors.py

```diff
@@ -51,3 +52,22 @@ class ObjectDeletedError(Error):
     def __init__(self, message: str, deleted_at: datetime.datetime):
         self.deleted_at = deleted_at
         super().__init__(message)
+
+
+class ClickhouseQueryError(Error):
+    """Raised when a query to Clickhouse fails."""
+
+    def __init__(
```

Comment:
you might want to include an optional user-safe message field that can be passed along.
you might want to include a unique query id